lib/functions_framework/function.rb

```diff
@@ -164,6 +181,66 @@ def call *args, globals: nil, logger: nil
       callable.call(*args)
     end
 
+    ##
```

Comment:
The "unwrapping" of the request, and building of an argument object to present to the callable, is currently done in `server.rb` in the App subclass. i.e. `HttpApp` creates a `Rack::Request` object, and `EventApp` handles decoding the cloud event. I think it's better to follow suit and do the same for your typed function: your `TypedApp` should handle the decoding of the json into a typed request object, rather than hacking the callable like you're doing below (where you're having to duplicate a bunch of logic, etc.) Thus, all you really need to do here in the `Function` class is add an attribute to store the `request_class` so that the app can access it.

Reply:
Reworked the implementation in terms of this suggestion, I think the code is greatly simplified. Thanks!